vault/login_mfa.go

```diff
@@ -286,6 +292,28 @@ func (i *IdentityStore) handleMFAMethodUpdateCommon(ctx context.Context, req *lo
 		}
 	}
 
+	// check if an MFA method configuration exists with that method name
+	if methodName != "" {
+		namedMfaConfig, err := b.MemDBMFAConfigByName(ctx, methodName)
+		if err != nil {
+			return nil, err
+		}
+		if namedMfaConfig != nil {
+			if mConfig == nil {
+				mConfig = namedMfaConfig
+			} else {
+				if mConfig.ID != namedMfaConfig.ID {
+					return nil, fmt.Errorf("a login MFA method configuration with the method name %s already exists", methodName)
+				}
+			}
+		}
+	}
+
+	// Updating the method config name
+	if mConfig != nil && methodName != "" && mConfig.Name != methodName {
+		mConfig.Name = methodName
```

Comment:
If we need this, how do you feel about moving this down below the `nil` mConfig handling, like this:
```
if methodName != "" {
  mConfig.Name = methodName
}
```

Alternatively, can we just set the struct's `methodName` after the `mConfig == nil` check/constructor, since the Config has `omitempty`?